### ISSUE TITLE
Move BasePaymentService to Typescript

### ISSUE THREAD

**Issue by srindom** [medusajs/medusa#1042]
We would like to move away from having interfaces in the separate `medusa-interfaces` NPM package and instead include them in `@medusajs/medusa`. Furthermore, the interfaces are right now classes that payment plugins etc. can extend from, we should move these to Typescript instead.

Eventually, all interfaces should be migrated to Typescript but for now we will begin with bringing the PaymentsService interface over. 

The first step will be to migrate the current [BasePaymentService](https://github.com/medusajs/medusa/blob/tax-api/packages/medusa-interfaces/src/payment-service.js) over to the core interfaces. A similar thing is happening in the `tax-api` branch right now: https://github.com/medusajs/medusa/blob/tax-api/packages/medusa/src/interfaces/tax-service.ts.

Once this is done we need to update all the existing payment plugins to use the interface. The plugins concerned are:
- [medusa-payment-adyen](https://github.com/medusajs/medusa/tree/master/packages/medusa-payment-adyen)
- [medusa-payment-klarna](https://github.com/medusajs/medusa/tree/master/packages/medusa-payment-klarna)
- [medusa-payment-manual](https://github.com/medusajs/medusa/tree/master/packages/medusa-payment-manual)
- [medusa-payment-paypal](https://github.com/medusajs/medusa/tree/master/packages/medusa-payment-paypal)
- [medusa-payment-stripe](https://github.com/medusajs/medusa/tree/master/packages/medusa-payment-stripe)

We will also have to evaluate if the current loader implementation should be updated when registering payment providers:
https://github.com/medusajs/medusa/blob/80e86ca08c6d825797bbb783484c21fb92a33a9c/packages/medusa/src/loaders/plugins.js#L218

cc: @adrien2p

**Comment by adrien2p** [comment]
@srindom coule you assign me the issue please

**Comment by srindom** [comment]
@adrien2p should we split this into more tickets? 

I am thinking that we can start with the payments interface and then create tickets for migrating each of the plugins, but let me know what you think?

**Comment by adrien2p** [comment]
Probably a hood idea, you know the payment better than me. As I mentioned, none of the providers use the service the same way did you see the same thing? Ill work on it from tomorrow on my spare time

**Comment by adrien2p** [comment]
But I think we should still keep a class to extend instead of an interface to implement. That way the loader keep working until it is migrated and then we can switch if the needs come

**Comment by adrien2p** [comment]
@srindom the first part is in the PR #1052
